run.sh

```diff
@@ -97,30 +96,51 @@ exec_to_load="$1"
 shift
 
 arguments="-m 2G -nographic -nodefaults "
-arguments+="-display none -serial stdio -device isa-debug-exit "
-arguments+="-fsdev local,security_model=passthrough,id=hvirtio0,path=$rootfs_9p "
-arguments+="-device virtio-9p-pci,fsdev=hvirtio0,mount_tag=fs0 "
-arguments+="-kernel $kvm_image "
-arguments+="-initrd $exec_to_load "
+arguments="${arguments}-display none -serial stdio -device isa-debug-exit "
+arguments="${arguments}-fsdev local,security_model=passthrough,id=hvirtio0,path=$rootfs_9p "
+arguments="${arguments}-device virtio-9p-pci,fsdev=hvirtio0,mount_tag=fs0 "
+arguments="${arguments}-kernel $kvm_image "
+arguments="${arguments}-initrd $exec_to_load "
 
 if test "$use_kvm" -eq 1; then
-    arguments+="-enable-kvm -cpu host "
+  file="/dev/kvm"
+  if ! [ -c "$file" ]; then
+    echo "$file does not exist, cannot run with KVM" 2>&1
+    exit 1
+  fi
+
+  # Check if the effective user has RW permission to /dev/kvm,
+  # if they haven't, temporarily add the user to the group that owns
+  # /dev/kvm, by default the KVM group. If that also fails, run QEMU
+  # as root, but with the -runas parameter to drop privileges.
+  if ! [ -r "$file" ] || ! [ -w "$file" ]; then
+    group_owner=$(stat -c "%G" $file)
+    if [ "$(stat -c "%A" $file | cut -c 5-6)" = "rw" ] && ! [ "$group_owner" = "$(id -un 0)" ];
+    then
+      sudo_prefix="sudo -g $group_owner -- "
```

Comment:
I will open up an issue to figure out when you couldn't run with `sudo` to fall back to using `qemu-system-x86_64` without KVM support.